pkgs/applications/office/paperless/default.nix

```diff
@@ -0,0 +1,222 @@
+{ stdenv
+, lib
+, writeScript
+, fetchFromGitHub
+, makeWrapper
+
+, python36
+, file
+, imagemagickBig # includes ghostscript
+, optipng
+, poppler
+, tesseract
+, unpaper
+}:
+
+## Usage
+
+# ${pkgs.paperless}/bin/paperless wraps manage.py
+
+# ${pkgs.paperless}/share/paperless/setup-env.sh can be sourced from a
+# shell script to setup a Paperless environment
+
+# paperless.withConfig is a convenience function to setup a
+# configured Paperless instance. Example:
+#
+# nix-build --out-link ./paperless -E '
+# (import <nixpkgs> {}).paperless.withConfig {
+#   dataDir = /tmp/paperless-data;
+#   config = {
+#     PAPERLESS_DISABLE_LOGIN = "true";
+#   };
+# }'
+#
+# Setup DB
+# ./paperless migrate
+#
+# Consume documents in dataDir/consume
+# ./paperless document_consumer --oneshot
+#
+# Start web interface
+# ./paperless runserver --noreload localhost:8000
+
+## WSGI with gunicorn
+# Use a shell script like this:
+# source ${paperless}/share/paperless/setup-env.sh
+# PYTHONPATH=$paperlessSrc gunicorn paperless.wsgi
+
+with lib;
+let
+  paperless = stdenv.mkDerivation rec {
```

Comment:
Can't we use `buildPythonApplication` here? This would allow us to specify `checkInputs` etc.
Of course, you'd have to override most of the phases.
I don't like that stuff like `pytest` is in the `pythonEnv` and hence not only available during the tests.

Reply:
I've separated the runtime and check inputs. 
The current use of `stdenv.mkDerivation` seems simpler to me than using `buildPythonApplication` (with `format = "other"`), so I haven't changed that.